src/utils/__tests__/gcsAPI.test.ts

```diff
@@ -7,235 +8,324 @@ import {
+          expect.objectContaining({
+            metadata,
+          })
+        );
+      });
+
+      it('errors if GCS upload goes sideways', async () => {
+        mockGCSUpload.mockImplementation(() => {
+          throw new Error('The squirrel got away!');
+        });
+
+        const { filename } = squirrelSimulatorArtifact;
+
+        await expect(
+          client.uploadArtifact(
+            squirrelSimulatorLocalPath,
+            squirrelSimulatorBucketPath
+          )
+        ).rejects.toThrowError(
+          `Encountered an error while uploading \`${filename}\``
+        );
+      });
+
+      it("doesn't upload anything in dry run mode", async () => {
+        process.env.DRY_RUN = 'true';
+
+        await client.uploadArtifact(
+          squirrelStatsLocalPath,
+          squirrelStatsBucketPath
+        );
+
+        expect(mockGCSUpload).not.toHaveBeenCalled();
+      });
+    }); // end describe('upload')
+
+    describe('download', () => {
+      it('calls the GCS library download method with the right parameters', async () => {
+        await client.downloadArtifact(
+          squirrelStatsArtifact,
+          tempDownloadDirectory
+        );
+
+        const { filename } = squirrelStatsArtifact;
+
+        expect(mockGCSDownload).toHaveBeenCalledWith({
+          destination: path.join(tempDownloadDirectory, filename),
+        });
+      });
+
+      it("errors if download directory doesn't exist", async () => {
+        // make sure it thinks it doesn't exist
+        syncExistsSpy.mockReturnValueOnce(false);
+
+        await expect(
+          client.downloadArtifact(
+            squirrelSimulatorArtifact,
+            tempDownloadDirectory
+          )
+        ).rejects.toThrowError(`directory does not exist!`);
+      });
+
+      it('errors if GCS download goes sideways', async () => {
+        mockGCSDownload.mockImplementation(() => {
+          throw new Error('The squirrel got away!');
+        });
+
+        const { filename } = squirrelSimulatorArtifact;
+
+        await expect(
+          client.downloadArtifact(
+            squirrelSimulatorArtifact,
+            tempDownloadDirectory
+          )
+        ).rejects.toThrowError(
+          `Encountered an error while downloading \`${filename}\``
+        );
+      });
+
+      it("doesn't upload anything in dry run mode", async () => {
+        process.env.DRY_RUN = 'true';
+
+        await client.downloadArtifact(
+          squirrelStatsArtifact,
+          tempDownloadDirectory
+        );
+
+        expect(mockGCSDownload).not.toHaveBeenCalled();
+      });
+    }); // end describe('download')
+
+    describe('listArtifactsForRevision', () => {
+      it('calls the GCS library getFiles method with the right parameters', async () => {
+        mockGCSGetFiles.mockReturnValue([[]]);
+
+        await client.listArtifactsForRevision(
+          dogsGHOrg,
+          squirrelRepo,
+          squirrelSimulatorCommit
+        );
```

Comment:
Perhaps you want an assert here that it actually returns an empty list?

Reply:
Isn't that just testing jest at that point, though? I've told jest, "Please return an empty list when I call this method," so asserting that it does so only tests whether jest is going to do what I asked, not anything to do with my code (since mocked return values ignore what happens inside the function entirely).

Reply:
Yeah, but you're mocking `getFiles` and then calling `listArtifactsForRevision`, that may ignore the result of `getFiles` in case of a bug or something.
So I guess what we also might want to test is that `listArtifactsForRevision` in this case returns the (processed) output of `getFiles`.

Reply:
Ohhhh - I had misunderstood who "it" was. Since my existing assertion is on the GCS method, I thought you were saying I should do

```javascript
mockGCSGetFiles.mockReturnValue([[]]);
...
expect(mockGCSGetFiles).toReturn([[]]);
```

Okay, sure, now I get what you're saying. Will add it.